src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java

```diff
@@ -111,15 +114,18 @@
      * @param overrideProps resolver for overriding properties
      * @param omitIgnoredModules {@code true} if ignored modules should be
      *         omitted
+     * @param threadModeSettings the thread mode configuration
      * @throws ParserConfigurationException if an error occurs
      * @throws SAXException if an error occurs
      */
     private ConfigurationLoader(final PropertyResolver overrideProps,
-                                final boolean omitIgnoredModules)
+                                final boolean omitIgnoredModules,
+                                final ThreadModeSettings threadModeSettings)
```

Comment:
please avoid changes in c-tors or static method `loadConfiguration`, create separate c-tor for you(for MT)
our plugins might use existing c-tors, so we should not force them to update their code.
maven plugin use static method - https://github.com/apache/maven-plugins/blob/trunk/maven-checkstyle-plugin/src/main/java/org/apache/maven/plugins/checkstyle/exec/DefaultCheckstyleExecutor.java#L377 , we can not recheck all plugins, there are too much of them.
If we have concern about alot of c-tors in class, we could minimize them on final steps of project.